clearwater-queue-manager.root/usr/share/clearwater/clearwater-queue-manager/scripts/check_restart_queue_state

```diff
@@ -18,14 +18,19 @@ then
   exit 1
 fi
 
+# Return 2 if any error occurs. Else return 1 if any node is queueing, and 0 if
+# restart completed.
 if [[ $use_single_restart_queue == "Y" ]] ; then
   /usr/share/clearwater/clearwater-queue-manager/env/bin/python /usr/share/clearwater/clearwater-queue-manager/scripts/check_queue_state.py "${management_local_ip:-$local_ip}" "$local_site_name" apply_config
 else
+  rc=0
   for apply_config_key in $( clearwater-etcdctl ls -p --recursive | grep apply_config_ | sort -n); do
     prefix="/clearwater/$local_site_name/configuration/"
     # Remove the prefix to get apply_config_<node_type> as etcd-key
     apply_config_key=${apply_config_key#$prefix}
 
     /usr/share/clearwater/clearwater-queue-manager/env/bin/python /usr/share/clearwater/clearwater-queue-manager/scripts/check_queue_state.py "${management_local_ip:-$local_ip}" "$local_site_name" "$apply_config_key"
+    rc=$(($?>rc?$?:rc))
```

Comment:
Urgh. This is pretty hard to follow! I suggest you either:

- Add a comment to explain what this line does
- Change this to something that's a bit less compact but easier to understand. How about just:
```
queue_state=$?
if [[ $queue_state > $rc ]]
then
  rc=$queue_state
fi
```
I think this will be a lot clearer for future code readers to understand.